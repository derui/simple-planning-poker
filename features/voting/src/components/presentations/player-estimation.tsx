import clsx from "clsx";

interface Props {
  name: string;
  estimated?: boolean;
}

const styles = {
  root: clsx("grid", "grid-rows-[auto_1fr]", "grid-cols-1", "m-3", "text-center", "max-w-12"),
  card: (estimated: boolean) =>
    clsx(
      "grid",
      "place-content-center",
      "h-20",
      "w-14",
      "rounded",
      "text-center",
      "items-center",
      "justify-center",
      "border",
      "border-orange-400",
      "text-orange-700",
      "transition-transform",
      {
        "bg-white": !estimated,
      },
      {
        "bg-orange-200": estimated,
        "[transform:rotateY(180deg)]": estimated,
      }
    ),

  eye: clsx("grid", "place-content-center"),
};

// eslint-disable-next-line func-style
export function PlayerEstimation(props: Props) {
  const { estimated } = props;

  return (
    <div className={styles.root}>
      <span>{props.name}</span>
      <span className={styles.card(estimated ?? false)} data-estimated={estimated ?? false}></span>
    </div>
  );
}
