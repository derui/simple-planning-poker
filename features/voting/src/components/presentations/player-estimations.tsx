import clsx from "clsx";
import { Skeleton } from "@spp/ui-skeleton";
import { PropsWithChildren } from "react";

interface Props {
  loading?: boolean;
}

const styles = {
  root: clsx(
    "grid",
    "grid-rows-1",
    "grid-cols-[auto_auto_auto_auto]",
    "auto-cols-min",
    "place-content-center",
    "gap-4",
    "min-h-12",
    "w-full"
  ),
  loading: clsx("grid", "grid-rows-1", "grid-cols-1", "place-content-center", "gap-4", "min-h-12", "w-full"),
};

/**
 * Container presentation for player estimations
 */
// eslint-disable-next-line func-style
export function PlayerEstimations({ loading, children }: PropsWithChildren<Props>) {
  if (loading) {
    return (
      <div className={styles.loading} data-loading="true">
        <Skeleton />
      </div>
    );
  }

  return <div className={styles.root}>{children}</div>;
}
