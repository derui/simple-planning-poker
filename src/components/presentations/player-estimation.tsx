import { ReactElement } from "react";
import classnames from "classnames";
import { BaseProps, generateTestId } from "../base";
import { iconize } from "../iconize";
import { UserMode } from "@/domains/game-player";
import { UserEstimationState } from "@/status/selectors/user-estimation";

interface Props extends BaseProps {
  userName: string;
  userMode: UserMode;
  displayValue: string;
  state: UserEstimationState;
}

const styles = {
  root: classnames("flex", "flex-col", "items-center"),
  card: (state: UserEstimationState) =>
    classnames(
      "flex",
      "flex-col",
      "h-20",
      "w-14",
      "rounded",
      "text-center",
      "items-center",
      "justify-center",
      "border",
      "border-primary-400",
      "m-3",
      "text-primary-500",
      "transition-transform",
      {
        "bg-white": state === "notSelected",
      },
      {
        "bg-primary-400": state === "estimated",
        "text-secondary1-200": state === "estimated",
        "[transform:rotateY(180deg)]": state === "estimated",
      },
      {
        "bg-white": state === "result",
        "text-primary-500": state === "result",
        "[transform:rotateY(0deg)]": state === "result",
      }
    ),

  eye: classnames("flex", "items-center", iconize("eye", { size: "l" }), "before:bg-primary-400"),
};

// eslint-disable-next-line func-style
export function PlayerEstimation(props: Props) {
  const gen = generateTestId(props.testid);

  let card: ReactElement;
  if (props.userMode === UserMode.inspector) {
    card = (
      <span className={styles.card("notSelected")} data-testid={gen("card")} data-mode="inspector">
        <span className={styles.eye} data-testid={gen("eye")}></span>
      </span>
    );
  } else {
    card = (
      <span className={styles.card(props.state)} data-testid={gen("card")} data-mode="normal" data-state={props.state}>
        {props.state === "result" ? props.displayValue : ""}
      </span>
    );
  }

  return (
    <div className={styles.root} data-testid={gen("root")}>
      <span>{props.userName}</span>
      {card}
    </div>
  );
}
