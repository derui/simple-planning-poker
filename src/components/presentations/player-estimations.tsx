import classnames from "classnames";
import { BaseProps, generateTestId } from "../base";
import { PlayerEstimation } from "./player-estimation";
import { Skeleton } from "./skeleton";
import { UserEstimationInfo } from "@/status/selectors/user-estimation";

interface Props extends BaseProps {
  estimations: UserEstimationInfo[];
  loading?: boolean;
}

const styles = {
  root: classnames("flex", "justify-around"),
};

// eslint-disable-next-line func-style
export function PlayerEstimations({ estimations, testid, loading }: Props) {
  const gen = generateTestId(testid);

  if (loading) {
    return (
      <div className={styles.root} data-testid={gen("root")}>
        <Skeleton testid={gen("loading")} />
      </div>
    );
  }

  const createUserEstimation = (props: UserEstimationInfo, index: number) => {
    return <PlayerEstimation key={index} testid={gen(props.userName)} {...props} />;
  };

  return (
    <div className={styles.root} data-testid={gen("root")}>
      {estimations.map((v, index) => createUserEstimation(v, index))}
    </div>
  );
}
