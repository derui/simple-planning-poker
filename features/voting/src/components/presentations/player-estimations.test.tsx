import { cleanup, render } from "@testing-library/react";
import { test, expect, afterEach } from "vitest";
import { PlayerEstimations } from "./player-estimations.js";
import { PlayerEstimation } from "./player-estimation.js";

afterEach(cleanup);

test("display a estimation", async () => {
  const { container } = render(
    <PlayerEstimations>
      <PlayerEstimation name="foo"></PlayerEstimation>
    </PlayerEstimations>
  );

  expect(container).toMatchSnapshot();
});

test("loading", async () => {
  const { container } = render(<PlayerEstimations loading />);

  expect(container.querySelector('[data-loading="true"]')).not.toBeNull();
});
