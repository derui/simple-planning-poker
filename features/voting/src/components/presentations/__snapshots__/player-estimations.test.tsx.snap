// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`display a estimation 1`] = `
<div>
  <div
    class="grid grid-rows-1 grid-cols-[auto_auto_auto_auto] auto-cols-min place-content-center gap-4 min-h-12 w-full"
  >
    <div
      class="grid grid-rows-[auto_1fr] grid-cols-1 m-3 text-center max-w-12"
    >
      <span>
        foo
      </span>
      <span
        class="grid place-content-center h-20 w-14 rounded text-center items-center justify-center border border-orange-400 text-orange-700 transition-transform bg-white"
        data-estimated="false"
      />
    </div>
  </div>
</div>
`;
